src/create-extension-pack.js

```diff
@@ -0,0 +1,184 @@
+        return;
+    }
+
+    const extPackSrcFolder = theiaExtension(extPackNameAndVersion);
+    if (!fs.existsSync(extPackSrcFolder)) {
+        await fs.mkdir(extPackSrcFolder);
+    }
+
+    const extPack = {};
+    extPack.name = packName;
+    extPack.displayName = packName;
+    extPack.description = 'Builtin extension pack associated to a version of vscode';
+    extPack.version = packVersion;
+    extPack.publisher = publisher;
+    extPack.categories = categories;
+    extPack.engines = { vscode: '^' + vscodeVer };
+    extPack.repository = repository;
+    extPack.extensionPack = await resolveExtensions();
+
+    if (extPack.extensionPack.length === 0) {
+        process.exitCode = 1;
+        console.error('Aborting: No extension was found available for this version: ' + packVersion);
+        return;
+    }
+
+    const packFolderPath = path.join(extPackSrcFolder, '..', extPackNameAndVersion)
+    const packJsonPath = path.join(packFolderPath, PACKAGE_JSON);
+    const licensePath = path.join(packFolderPath, 'LICENSE.txt');
+    const readmePath = path.join(packFolderPath, 'README.md');
+
+    fs.writeFileSync(packJsonPath, JSON.stringify(extPack, null, 2), 'utf-8');
+    console.log('Generated ' + PACKAGE_JSON + ' file at: ' + packJsonPath);
+    fs.writeFileSync(licensePath, genLicense());
+    fs.writeFileSync(readmePath, genReadme());
+
+    await vsce.createVSIX({
+        'cwd': packFolderPath,
+        'packagePath': dist(),
+        'useYarn': true
+    });
+
+    async function resolveExtensions() {
+        const extensionsArr = [];
+        for (const extension of fs.readdirSync(extensions())) {
+            const extDataPath = extensions(extension, PACKAGE_JSON);
+            if (!fs.existsSync(extDataPath)) {
+                console.log('No ' + PACKAGE_JSON + ' found for: ' + extension);
+                continue;
+            }
+
+            const content = fs.readFileSync(extDataPath, 'utf-8');
+            const extData = JSON.parse(content);
+
+            const extVsixPath = dist(extData.name + '-' + packVersion + '.vsix');
+            const available = await isAvailable(extVsixPath, extData.name, packVersion);
+            if (!available) {
+                console.log("Skipping extension, i.e. .vsix is not found and " +
+                    "neither published in the registry : " + extVsixPath);
+                continue;
+            }
+
+            const extensionId = extData.publisher + '.' + extData.name;
+            console.log('Adding: ' + extensionId);
+            extensionsArr.push(extensionId);
+        }
+        return Promise.resolve(extensionsArr);
+    }
+
+    async function isAvailable(extVsixPath, extensionName, extensionVersion, namespace = 'vscode') {
+        if (fs.existsSync(extVsixPath)) {
+            return Promise.resolve(true);
+        }
+
+        return isPublished(extensionVersion, extensionName, namespace);
+    }
+})();
+
+function genLicense() {
```

Comment:
We can be explicit in the naming:

```suggestion
function generateLicense() {
```

Reply:
Done !